homeassistant/components/climate/radiotherm.py

```diff
@@ -219,6 +219,20 @@ def is_away_mode_on(self):
         """Return true if away mode is on."""
         return self._away
 
+    @property
+    def state(self):
```

Comment:
This isn't allowed. Platforms should not overwrite the base climate component state implementation.

See https://github.com/home-assistant/architecture/issues/22 for the future of the climate component.

Reply:
Am I correct in my reading that what should happen is that the derived class should be setting is_on to true/false and that's the correct fix in this case?

Reply:
Yes, for now, we can show on/off state by using `is_on` property. :+1:

Reply:
Also - I'm confused.  The base class does this for state()
```
        if self.is_on is False:
            return STATE_OFF
        if self.current_operation:
            return self.current_operation
```
but what if current operation is auto?  Then state doesn't indicate what's actually happening.

Reply:
The current meaning of state isn't consistent. But we have to follow the current architecture, until it's reworked.

Reply:
Sorry - not sure I understand.  It's not very clear in the code in most places but from what I can see in the base class, "current_operation" is what I would call the mode: what you want to happen if the temperature is outside the nominal range.  So in __init__, if is_on == True, the mode is the state that is returned.  But some/many thermostats have an AUTO mode where what will happen will depend on the temp bounds so the base class state method (that I shouldn't override) is wrong.  The method I proposed in this pull request for radiotherm does work correctly in that case so I'm not sure why we shouldn't just merge it.  If someone is going to rework the whole climate API we can update everything when that's complete but at least it will work correctly now.

Reply:
OK - Updated to use is_on for now.  I can help with rewriting it more fully once the climate architecture is settled on.